src/core/CMakeLists.txt

```diff
@@ -101,6 +123,13 @@ set(libraryname idyntree-core)
 
 add_library(${libraryname} ${IDYNTREE_CORE_EXP_SOURCES} ${IDYNTREE_CORE_EXP_HEADERS} ${IDYNTREE_CORE_EXP_PRIVATE_INCLUDES})
 
+if (DEFINED ${CMAKE_COMPILER_IS_GNUCXX} 
```

Comment:
O_o . 
Then I would probably drop the defined, as that variable is available since CMake 2.8.11 : https://cmake.org/cmake/help/v2.8.11/cmake.html#variable:CMAKE_COMPILER_IS_GNULANG .